router/handler_edge_ctrl/apiSessionAdded.go

```diff
@@ -17,22 +17,43 @@
 package handler_edge_ctrl
 
 import (
+	"encoding/json"
+	"errors"
+	"fmt"
 	"github.com/golang/protobuf/proto"
 	"github.com/michaelquigley/pfxlog"
 	"github.com/openziti/edge/controller/env"
-	"github.com/openziti/edge/router/internal/fabric"
+	"github.com/openziti/edge/controller/sync_strats"
 	"github.com/openziti/edge/pb/edge_ctrl_pb"
+	"github.com/openziti/edge/router/internal/fabric"
 	"github.com/openziti/foundation/channel2"
+	"sync"
+	"time"
 )
 
 type apiSessionAddedHandler struct {
-	sm fabric.StateManager
+	control     channel2.Channel
+	sm          fabric.StateManager
+	syncTracker *apiSessionSyncTracker
+	reqChan     chan *apiSessionAddedWithState
+	syncReady   chan []*edge_ctrl_pb.ApiSession
+	syncFail    chan error
 }
 
-func NewApiSessionAddedHandler(sm fabric.StateManager) *apiSessionAddedHandler {
-	return &apiSessionAddedHandler{
-		sm: sm,
+func NewApiSessionAddedHandler(sm fabric.StateManager, control channel2.Channel) *apiSessionAddedHandler {
+	handler := &apiSessionAddedHandler{
+		control:   control,
+		sm:        sm,
+		reqChan:   make(chan *apiSessionAddedWithState, 100),
+		syncReady: make(chan []*edge_ctrl_pb.ApiSession, 0),
+		syncFail:  make(chan error, 0),
 	}
+
```

Comment:
Are these getting shutdown when we lose/re-establish a connection to the controller?

Reply:
Fixed.